ldap/admin/src/logconv.py

```diff
@@ -608,246 +621,617 @@ def _init_regexes(self):
-        if len(self.result.optime_duration) > self.size_limit:
-            heapq.heappop(self.result.optime_duration)
+        return default
 
-        # Total result times
-        self.result.total_etime = self.result.total_etime + etime
-        self.result.total_wtime = self.result.total_wtime + wtime
-        self.result.total_optime = self.result.total_optime + optime
+    def build_op_scope_key(self, conn_scope_key, op_id):
+        """
+        Build a flat operation scoped key from a connection scoped key and operation ID.
 
-        # Statistic reporting
-        self.result.etime_stat = round(self.result.etime_stat + float(etime), 8)
+        Args:
+            conn_scope_key (str | tuple): Either a legacy tuple (restart, conn_id),
+                                        or a new-format string like "1231231-2".
+            op_id (int | str): The operation ID associated with this log entry.
 
-        if err is not None:
-            self.result.error_freq[err] += 1
+        Returns:
+            tuple: A flattened tuple of the form (restart, conn_id, op_id) that
+                uniquely identifies an operation.
 
-        # Check for internal operations based on either conn_id or internal flag
-        if 'Internal' in conn_id or internal:
-            self.operation.counters['internal'] +=1
+        Raises:
+            ValueError: If the input format is invalid or op_id is None.
+        """
+        if op_id is None:
+            raise ValueError("op_id cannot be None")
 
-        # Process result notes if present
-        NOTE_TYPES = {'A', 'U', 'F', 'M', 'P'}
-        if notes in NOTE_TYPES:
-            note_dict = self.result.notes[notes]
-            self.result.counters[f'notes{notes}'] += 1
-
-            # Exclude VLV
-            if restart_conn_op_key not in self.vlv.rst_con_op_map:
-                # Construct the notes dict
-                note_dict = self.result.notes[notes]
-                note_entry = note_dict.setdefault(restart_conn_op_key, {})
-                note_entry.update({
-                    'time': timestamp,
-                    'etime': etime,
-                    'nentries': nentries,
-                    'ip': self.connection.restart_conn_ip_map.get(restart_conn_key, 'Unknown IP'),
-                    'bind_dn': self.bind.restart_conn_dn_map.get(restart_conn_key, 'Unknown DN')
+        # Convert op_id to int if needed
+        op_id = int(op_id)
+
+        # Legacy format
+        if isinstance(conn_scope_key, tuple) and len(conn_scope_key) == 2:
+            restart, conn_id = conn_scope_key
+            return (restart, conn_id, op_id)
+
+        # New format: string like "1231231-2"
+        if isinstance(conn_scope_key, str) and '-' in conn_scope_key:
+            parts = conn_scope_key.split('-')
+            if len(parts) == 2 and all(part.isdigit() for part in parts):
+                restart, conn_id = map(int, parts)
+                return (restart, conn_id, op_id)
```

Comment:
That feels a bit fragile and error prone.
Can we standardize on tuples? (and convert all keys to tuples at the entry point)

Reply:
It does look a bit clunky, now we normalise the key at the entry point